### PR TITLE
Reactions & Retractions

### PR DESCRIPTION
**Pull request by nesium**
This PR adds support for sending and receiving message reactions as well as retracting messages. It also fixes the broken unit and UI tests.

I've updated the use of Character for reactions with a custom Reactions struct for future compatibility. I'm assuming that at some point we'll allow people their own emojis. It also makes the intent clearer in call sites especially when calling into closures where we don't have argument labels.

### CODE REVIEW

**Comment by RemiBardon** [comment]
I will have a look first thing in the afternoon

File: Prose/ProseLib/Sources/ProseUI/View+OnKeyDown.swift [hunk 1]
```diff
@@ -63,7 +63,7 @@ public extension View {
   ///         we ensure ``KeyAwareView`` stays first responder as long as we need it to, and it correctly
   ///         receives the key events.
   func onKeyDown(perform onEvent: @escaping (KeyEvent) -> Void) -> some View {
-    KeyAwareView(rootView: self, onEvent: onEvent)
+    KeyAwareView(rootView: self, onEvent: onEvent).accessibilityElement(children: .contain)
```
[the remark this answers is not in the document]
Comment (nesium) [reply Prose/ProseLib/Sources/ProseUI/View+OnKeyDown.swift:66]: Yes

[the remark this answers is not in the document]
Comment (RemiBardon) [reply Prose/ProseLib/Sources/ProseUI/View+OnKeyDown.swift:66]: Ok, I just hope it make sense in terms of AX then

File: Prose/ProseLib/Sources/ProseCoreTCA/Types/Message.swift [hunk 2]
```diff
@@ -89,23 +93,45 @@ extension MessageKind {
...
+    for (reaction, _) in self.reactions {
+      self.reactions[reaction]?.remove(jid)
+      if self.reactions[reaction]?.isEmpty == true {
+        self.reactions.removeValue(forKey: reaction)
+      }
+    }
+    for reaction in reactions {
+      self.reactions[reaction, default: []].insert(jid)
+    }
+  }
```
[the remark this answers is not in the document]
Comment (nesium) [reply Prose/ProseLib/Sources/ProseCoreTCA/Types/Message.swift:120]: It's used in the code and tested even. More info [here](https://xmpp.org/extensions/xep-0444.html#updating-reactions).

[the remark this answers is not in the document]
Comment (RemiBardon) [reply Prose/ProseLib/Sources/ProseCoreTCA/Types/Message.swift:120]: I had seen you tested it, but I didn't see it in `ProseClient`. Thank you for the XEP link 🙏🏼

**Review by RemiBardon — APPROVED** [review]
Everything looks good, thank you 🙏🏼

**Pull request merged**